src/workflows-app/pages/SubmissionHistory.js

```diff
@@ -0,0 +1,333 @@
+import _ from 'lodash/fp';
+import { Fragment, useRef, useState } from 'react';
+import { div, h, h2 } from 'react-hyperscript-helpers';
+import { AutoSizer } from 'react-virtualized';
+import { ButtonOutline, Clickable, Link } from 'src/components/common';
+import { centeredSpinner, icon } from 'src/components/icons';
+import { MenuButton } from 'src/components/MenuButton';
+import { MenuTrigger } from 'src/components/PopupTrigger';
+import { FlexTable, paginator, Sortable, tableHeight, TextCell } from 'src/components/table';
+import { Ajax } from 'src/libs/ajax';
+import { Apps } from 'src/libs/ajax/leonardo/Apps';
+import colors from 'src/libs/colors';
+import * as Nav from 'src/libs/nav';
+import { notify } from 'src/libs/notifications';
+import { useCancellation, useOnMount } from 'src/libs/react-utils';
+import { getUser } from 'src/libs/state';
+import * as Utils from 'src/libs/utils';
+import { resolveRunningCromwellAppUrl } from 'src/workflows-app/utils/app-utils';
+import {
+  AutoRefreshInterval,
+  getDuration,
+  isRunSetInTerminalState,
+  loadAllRunSets,
+  makeStatusLine,
+  statusType,
+} from 'src/workflows-app/utils/submission-utils';
+import { wrapWorkflowsPage } from 'src/workflows-app/WorkflowsContainer';
+
+export const BaseSubmissionHistory = ({ name, namespace, workspace }, _ref) => {
+  // State
+  const [sort, setSort] = useState({ field: 'submission_timestamp', direction: 'desc' });
+  const [pageNumber, setPageNumber] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState(50);
+  const [runSetsData, setRunSetData] = useState();
+  const [runSetsFullyUpdated, setRunSetsFullyUpdated] = useState();
+  const [loading, setLoading] = useState(false);
+
+  const signal = useCancellation();
+  const scheduledRefresh = useRef();
+  const workspaceId = workspace.workspace.workspaceId;
+
+  // helper for auto-refresh
+  const refresh = Utils.withBusyState(setLoading, async () => {
+    try {
+      const loadedRunSetData = await loadAllRunSets(signal, workspaceId);
```

Comment:
It seems the `loadAllRunSets` method might need to be updated to use the `workflowsAppStore` to fetch the CBAS proxy url.